### ISSUE TITLE
Cache zip file to allow for offline execution...

### ISSUE THREAD

**Issue by sean-** [apple/pkl-go#42]
When running `pkl-gen-go`, I shouldn't need to be online or have a stable internet connection.

In my `.pkl` file, I have:

```
import "package://pkg.pkl-lang.org/pkl-go/pkl.golang@0.6.0#/go.pkl"
```

Which downloads `https://github.com/apple/pkl-go/releases/download/pkl.golang@0.6.0/pkl.golang@0.6.0.zip` every time.  Can this file be cached since I'm using a stable tagged version of pkl?

```
panic: –– Pkl Error ––
Exception when making request `GET https://github.com/apple/pkl-go/releases/download/pkl.golang@0.6.0/pkl.golang@0.6.0.zip`:
Operation timed out


goroutine 1 [running]:
main.init.1()
    /Users/seanc/go/pkg/mod/github.com/apple/pkl-go@v0.6.0/cmd/pkl-gen-go/pkl-gen-go.go:245 +0x62c
```

**Comment by holzensp** [comment]
It's a bug if it isn't cached. Are you seeing this in CI? In some sort of build system that attempts to make things hermetic? Pkl's default cache is `~/.pkl/cache`, but you can set it explicitly with `--cache-dir`. I just tried it and failed to reproduce.

**Comment by sean-** [comment]
I looked for that type of option to start, and noticed that in [`WithDefaultCacheDir`](https://github.com/apple/pkl-go/blob/acc815a52953f160d073184c82f2b6eb1554e738/pkl/evaluator_options.go#L245-L253), there isn't a way to override its behavior:

```go
// WithDefaultCacheDir sets the cache directory to Pkl's default location.
// It panics if the home directory cannot be determined.
var WithDefaultCacheDir = func(opts *EvaluatorOptions) {
  dirname, err := os.UserHomeDir()
  if err != nil {
    panic(err)
  }
  opts.CacheDir = path.Join(dirname, ".pkl/cache")
}
```

```
$ pkl-gen-go --help | grep -i cache
$ pkl-gen-go --version
0.6.0
```

I didn't see a way where these flags were being magically plumbed through to the `CacheDir` evaluator setting in `pkl-gen-go`, either.  Does this flag only exist internally w/in Apple and not in the OSS version of `pkl-gen-go`?

As a [`plz`](http://please.build) user, this would be a pretty amazing quality of life improvement because `os.UserHomeDir` changes every time a rule is updated to keep builds hermetic, just like `bazel`.  I've unblocked myself with the following workaround, but it's pretty unsavory:

```
cmd = "env HOME=$(echo $HOME | perl -p -e 's#plz-out(.*)#plz-out/.pkl-gen-go/v0.6.0#') pkl-gen-go ..."
```

If a `--cache-dir` flag existed, this would be preferred.

**Comment by holzensp** [comment]
Closed by #59
